bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/LightThingHandler.java

```diff
@@ -0,0 +1,322 @@
+                "state");
+
+        String json = gson.toJson(newLightState);
+        logger.trace("Sending {} to light {} via {}", json, config.id, url);
+
+        asyncHttpClient.put(url, json, bridgeConfig.timeout).thenAccept(v -> {
+            lastCommandTimestamp = System.currentTimeMillis();
+            lastCommand = newLightState;
+            logger.trace("Result code={}, body={}", v.getResponseCode(), v.getBody());
+        }).exceptionally(e -> {
+            logger.debug("Sending command {} to channel {} failed:", command, channelUID, e);
+            return null;
+        });
+    }
+
+    @Override
+    protected @Nullable LightMessage parseStateResponse(AsyncHttpClient.Result r) {
+        if (r.getResponseCode() == 403) {
+            return null;
+        } else if (r.getResponseCode() == 200) {
+            return gson.fromJson(r.getBody(), LightMessage.class);
+        } else {
+            throw new IllegalStateException("Unknown status code " + r.getResponseCode() + " for full state request");
+        }
+    }
+
+    @Override
+    protected void processStateResponse(@Nullable LightMessage stateResponse) {
+        if (stateResponse == null) {
+            return;
+        }
+        messageReceived(config.id, stateResponse);
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private void valueUpdated(String channelId, LightState newState) {
+        Integer bri = newState.bri;
+        Boolean on = newState.on;
+
+        switch (channelId) {
+            case CHANNEL_SWITCH:
+                if (on != null) {
+                    updateState(channelId, OnOffType.from(on));
+                }
+                break;
+            case CHANNEL_COLOR:
+                Double @Nullable [] xy = newState.xy;
+                Integer hue = newState.hue;
+                Integer sat = newState.sat;
+                if (hue != null && sat != null && bri != null) {
+                    updateState(channelId,
+                            new HSBType(new DecimalType(hue / HUE_FACTOR), toPercentType(sat), toPercentType(bri)));
+                } else if (xy != null && xy.length == 2) {
+                    updateState(channelId, HSBType.fromXY(xy[0].floatValue(), xy[1].floatValue()));
+                }
+                break;
+            case CHANNEL_BRIGHTNESS:
+                if (bri != null && on != null && on) {
+                    updateState(channelId, toPercentType(bri));
+                } else {
+                    updateState(channelId, OnOffType.OFF);
+                }
+                break;
+            case CHANNEL_COLOR_TEMPERATURE:
+                Integer ct = newState.ct;
+                if (ct != null) {
+                    updateState(channelId, new DecimalType(scaleColorTemperature(ct)));
+                }
+                break;
+            case CHANNEL_POSITION:
+                if (bri != null) {
+                    updateState(channelId, toPercentType(bri));
```

Comment:
This is the same as brightness?

Reply:
Yes. Blinds are modelled as lights in the deconz REST API and the position is set and reported as brightness.